src/games/strategy/triplea/delegate/Matches.java

```diff
@@ -121,6 +121,13 @@ public boolean match(final Unit unit) {
       return (ua.getIsCombatTransport() && ua.getIsSea());
     }
   };
+  public static final Match<Unit> UnitIsDefenselessTransport = new Match<Unit>() {
+    @Override
+    public boolean match(final Unit unit) {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      return (ua.getTransportCapacity() != -1 && ua.getIsSea() && ua.getRawDefense() == 0);
```

Comment:
This logic isn't 100% fool proof. You could have a sea transport unit that gives defense support to other sea transports that would make it so they ended up with more than 0 defense. You need to check game properties as I'm almost positive v3 has a property around defenseless transports.